mmdeploy/codebase/mmflow/ops/corr_lookup.py

```diff
@@ -0,0 +1,114 @@
+            pixels. If set to False, they are instead considered as referring
+            to the corner points of the input’s corner pixels, making the
+            sampling more resolution agnostic. Default to False.
+        scale (bool): Whether scale the grid values in the range of [-1, 1].
+            Defaults to True.
+
+    Returns:
+        Tensor: The output tensor using input feature values and pixel
+            locations from grid
+    """
+    H, W = feat.shape[-2:]
+    # if grid.shape[-1] != 2:
+    #     grid = grid.permute(0, 2, 3, 1)
+    # if grid.shape[-1].equal(grid.shape[-1] - grid.shape[-1] + 2):
+    #     pass
+    # else:
+    #     grid = grid.permute(0, 2, 3, 1)
+    if scale:
+        # grid[:, :, :, 0] = grid[:, :, :, 0] * 2. / max(W - 1, 1) - 1.
+        # grid[:, :, :, 1] = grid[:, :, :, 1] * 2. / max(H - 1, 1) - 1.
+        grid[:, :, :, 0] = grid[:, :, :, 0] * 2. / (W - 1) - 1.
+        grid[:, :, :, 1] = grid[:, :, :, 1] * 2. / (H - 1) - 1.
+    return F.grid_sample(feat, grid, mode, padding_mode, align_corners)
+
+
+def coords_grid(batch: int, xx: Tensor, yy: Tensor) -> Tensor:
+    """Coordinate grid.
+
+    Args:
+        batch (int): The batch size of feature.
+        xx (Tensor): 1-D tensor of size W with values from the interval
+            [0, W-1].
+        yy (Tensor): 1-D tensor of size H with values from the interval
+            [0, H-1].
+
+    Returns:
+        Tensor: Tensor of shape (batch, 2, H, W) with values of items'
+            coordinate.
+    """
+    coords = torch.meshgrid(yy, xx)
+    coords = torch.stack(coords[::-1], dim=0).float()
+
+    return coords[None].repeat(batch, 1, 1, 1)  # shape(batch, 2, H, W)
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmflow.ops.corr_lookup.CorrLookup.forward')
+def corr_lookup_forward(ctx,self, corr_pyramid: Sequence[Tensor], flow: Tensor) -> Tensor:
+    """Forward function of Correlation lookup.
+
+    Args:
+        corr_pyramid (Sequence[Tensor]): Correlation pyramid.
+        flow (Tensor): Current estimated optical flow.
+
+    Returns:
+        Tensor: Feature map by indexing from the correlation pyramid.
+    """
+    B, _, H, W = flow.shape
+    xx = torch.arange(0, W, device=flow.device)
+    yy = torch.arange(0, H, device=flow.device)
+    grid = coords_grid(B, xx, yy) + flow  # shape N, 2, H, W
+    grid = grid.permute(0, 2, 3, 1)  # shape N, H, W, 2
+
+    dx = torch.linspace(
+        -self.r, self.r, 2 * self.r + 1, device=flow.device)
+    dy = torch.linspace(
+        -self.r, self.r, 2 * self.r + 1, device=flow.device)
+    delta = torch.stack(torch.meshgrid(dy, dx), axis=-1)
+    delta_lvl = delta.view(1, 2 * self.r + 1, 2 * self.r + 1, 2)
+
+    out_corr_pyramid = []
+    for i, corr in enumerate(corr_pyramid):
+        centroid_lvl = grid.reshape(B * H * W, 1, 1, 2) / 2 ** i
+        coords_lvl = centroid_lvl + delta_lvl
+
+        corr = rewrite_bilinear_sample(corr, coords_lvl, self.mode,
```

Comment:
If this rewriter just replace `bilinear_sample` with `rewrite_bilinear_sample`, why not just create a function rewriter for `bilinear_sample`?